packages/program-boilerplate/src/utils.ts

```diff
@@ -64,3 +64,53 @@ export function getGoalAnalyticTimestamp(trigger: any): number {
 
   return purchaseEvent ? purchaseEvent.dateTriggered - 1 : trigger.time;
 }
+
+/**
+ * Translates a string type into its proper JavaScript type.
+ * Only supports conversion to number, boolean, null and undefined values.
+ * Objects and arrays are not supported.
+ *
+ * @example
+ * "29" => 29 (number)
+ * @example
+ * "3.1415" => 3.1415 (number)
+ * @example
+ * "true" => true (boolean)
+ * @example
+ * "asdf" => "asdf" (string)
+ * @example
+ * NaN => NaN (number)
+ * @example
+ * undefined => undefined
+ *
+ * @param {string} val The value to test
+ * @return {any} the computed value
+ */
+export function inferType(val: string): any {
+  if (/^(-|\+)?([0-9]+(\.[0-9]+)?)$/.test(val)) {
```

Comment:
I guess you could note trying to parse the number directly won't work here because something like "999a" would become 999, which would be undesirable in most scenarios.